### PR TITLE
Spark: Remove ImmutableMap from SparkPartition (spark 3.2)

### PR DESCRIPTION
**Pull request by dubeme**
Issue #3586 shows situation where Kyro is unable to serialize the SparkPartition object (@rdblue points it out from the stack trace). This commit replaces the ImmutableMap with HashMap

I tested this manually, by creating a .jar, then running a local spark application. I unfortunately don't have the deep understanding to add the Unit tests for the `add_files` action.

### CODE REVIEW

File: spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java [hunk 1]
```diff
@@ -647,7 +647,7 @@ private static void deleteManifests(FileIO io, List<ManifestFile> manifests) {
     private final String format;
 
     public SparkPartition(Map<String, String> values, String uri, String format) {
-      this.values = ImmutableMap.copyOf(values);
+      this.values = new HashMap<>(values);
```
Comment (rdblue) [inline spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java:650]: Can you use `Maps.newHashMap`? We prefer the Guava factory methods to individual implementations.

Also, can you add a test for this that does a round-trip serialization using Kryo?

**Comment by rdblue** [comment]
Thanks for working on this, @dubeme!

**Review by kbendick — COMMENTED** [review]
Thanks for the contribution @dubeme!

For the serialization test, it would probably be sufficient to simply instantiate one and then ensure it can be round tripped (same object before and after) via Kryo serde.

There's a helper function for this which you can see in this similar test: https://github.com/apache/iceberg/blob/f5a753791f4dc6aca78569a14f731feda9edf462/spark/v3.2/spark/src/test/java/org/apache/iceberg/TestFileIOSerialization.java#L81-L92

**Comment by jackye1995** [comment]
Hi @dubeme, we are approaching the release of 0.13, and this is a critical bug that needs to be merged before the branch cut. Would you have time to update this PR? If not I can put up the same fix and mark you as coauthor.

**Comment by kbendick** [comment]
> Hi @dubeme, we are approaching the release of 0.13, and this is a critical bug that needs to be merged before the branch cut. Would you have time to update this PR? If not I can put up the same fix and mark you as coauthor.

Agreed. I’d say we give them a few more days and then we mark them as co-author. I’d be happy to do it as well if you’re busy. Let me know!

**Comment by rdblue** [comment]
Thanks, @dubeme! This was fixed by #3667.

**Pull request closed without merging**